### PR TITLE
Update JS pipeline generator to add new variable group

### PR DESCRIPTION
**Pull request by praveenkuttappan**
JS live test needs an additional variable group for JS specific variables. This PR will configure this additional variable group for JS.

### CODE REVIEW

**Comment by benbp** [comment]
Part of me wants to not merge this so we can try to stop people from consuming this variable group entirely on a case by case basis...

**Comment by praveenkuttappan** [comment]
> Part of me wants to not merge this so we can try to stop people from consuming this variable group entirely on a case by case basis...

Are we planning to move to single variable group with all required keys?

**Comment by benbp** [comment]
@praveenkuttappan the recommended approach is to avoid a 1:1 mapping between values and secrets, in favor of a config file hierarchy in as few keyvaults as possible. This makes secret self service easier for devs and reduces all the locations and service principals that need to be tracked when variable groups proliferate. See https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/389/Configuring-Tests?anchor=configuring-subscriptions-and-cloud-environments for docs on this.

File: eng/pipelines/pipeline-generation.yml [hunk 1]
```diff
@@ -68,7 +70,9 @@ jobs:
           --variablegroup ${{ parameters.Release_Secrets_for_GitHubIO_Publishing }}
           --variablegroup ${{ parameters.Secrets_for_DevOps_Notification_Configuration_and_User_Resolution }}
           --variablegroup ${{ parameters.APIReview_AutoCreate_Configurations }}
-        TestOptions: '--variablegroup ${{ parameters.Secrets_for_Resource_Provisioner }}'
+        TestOptions: >-
+          --variablegroup ${{ parameters.Secrets_for_Resource_Provisioner }}
+          --variablegroup ${{ parameters.Test_Secrets_for_JS_Live_Tests }}
```
Comment (weshaggard) [inline eng/pipelines/pipeline-generation.yml:75]: If we do take this change we will also need to update https://github.com/Azure/azure-sdk-for-js/blob/main/eng/pipelines/prepare-pipelines.yml#L10

**Comment by weshaggard** [comment]
@praveenkuttappan I assume this is in response to https://github.com/Azure/azure-sdk-tools/issues/3741. If so please read over that discussion. I'm also hesitant to blindly add this variable group to all test pipelines.

**Pull request closed without merging**